utils/dbutils.py

```diff
@@ -0,0 +1,145 @@
+    dbConn = os.getenv("GEM_ONLINE_DB_CONN")
+    df_gemView = pd.read_sql(query, con=(dbConn+dbName))
+
+    if debug:
+        df_gemView.info()
+        print("Read {0} rows from view {1}".format(df_gemView.shape[0],view))
+        pass
+
+    if vfatList is not None:
+        # First warn the user which VFATs are *not* found
+        if len(vfatList) != df_gemView.shape[1]:
+            printYellow("Length of returned view does not match length of input vfat List")
+            vfatsNotFound = [ "0x{:x}".format(chipId) for chipId in vfatList if "0x{:x}".format(chipId) not in list(df_gemView['vfat3_ser_num'])]
+            printYellow("VFATs not found: {0}".format(vfatsNotFound))
+            pass
+        
+        # Then add a 'vfatN' column to the output df; this increases row # to len(vfatList)
+        df_gemView = joinOnVFATSerNum(vfatList,df_gemView)
+        
+        pass
+
+    return df_gemView
+
+def getVFAT3CalInfo(vfatList, debug=False):
+    """
+    Gets from GEM_VFAT3_PROD_SUMMARY_V_RH view a subset of data that is necessary
+    for VFAT calibration.  Specifically a pandas dataframe will be returned with
+    only the following columns:
+
+        ['vfatN','vfat3_ser_num', 'vfat3_barcode', 'iref', 'adc0m', 'adc1m', 'adc0b', 'adc1b', 'cal_dacm', 'cal_dacb']
+
+    vfatList    - list of VFAT Chip ID's.
+    debug       - Prints additional info if true
+    """
+
+    df_vfatCalInfo = getVFAT3ProdSumView(vfatList, debug)
+
+    return df_vfatCalInfo[['vfatN','vfat3_ser_num', 'vfat3_barcode', 'iref', 'adc0m', 'adc1m', 'adc0b', 'adc1b', 'cal_dacm', 'cal_dacb']]
+
+def getVFAT3ConfView(vfatList, debug=False):
+    """
+    Gets the GEM_VFAT3_CHIP_CONF_V_RH view in the GEM DB for a list of input VFATs.
+
+    Returns a pandas dataframe object storing the data read from the DB
+
+    vfatList    - list of VFAT Chip ID's.
+    debug       - Prints additional info if true
+    """
+
+    return getGEMDBView("GEM_VFAT3_CHIP_CONF_V_RH",vfatList,debug)
+
+def getVFAT3ProdSumView(vfatList, debug=False):
+    """
+    Gets the GEM_VFAT3_PROD_SUMMARY_V_RH view in the GEM DB for a list of input VFATs.
+
+    Returns a pandas dataframe object storing the data read from the DB
+
+    vfatList    - list of VFAT Chip ID's.
+    debug       - Prints additional info if true
+    """
+
+    return getGEMDBView("GEM_VFAT3_PROD_SUMMARY_V_RH",vfatList,debug)
+
+def getVFATFilter(vfatList):
+    """
+    Returns a string that can be used as a filter in an SQL query.
+
+    vfatList - list of VFAT Chip ID's
+    """
+
+    strRetFilter = " WHERE ("
+    for idx, vfatID in enumerate(vfatList):
+        if idx == 0:
+            strRetFilter += " data.VFAT3_SER_NUM='0x{:x}'".format(vfatID)
+        else:
+            strRetFilter += " OR data.VFAT3_SER_NUM='0x{:x}'".format(vfatID)
+            pass
+        pass
+    strRetFilter += " )\n"
+
+    return strRetFilter
+
+def joinOnVFATSerNum(vfatList, dfGEMView):
```

Comment:
Here and below you use `dfGEMView` variable while above (e.g. l60 it is `df_gemView`)

Reply:
#SplittingHairs

Reply:
unified to `dfGEMView`